File: ActivityLauncherApp/src/main/java/de/szalkowski/activitylauncher/IconPickerFragment.java
package de.szalkowski.activitylauncher;

import android.content.Context;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.GridView;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;

public class IconPickerFragment extends Fragment implements IconListAsyncProvider.Listener<IconListAdapter> {
    protected GridView grid;

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.icon_picker, null);

        this.grid = (GridView) view;
        this.grid.setOnItemClickListener((view1, item, index, id) ->
                Toast.makeText(getActivity(), view1.getAdapter().getItem(index).toString(), Toast.LENGTH_SHORT).show());

        return view;
    }

    @Override
    public void onAttach(@NonNull Context activity) {
        super.onAttach(activity);

        IconListAsyncProvider provider = new IconListAsyncProvider(this.getActivity(), this);
        provider.execute();
    }

    @Override
    public void onProviderFinished(AsyncProvider<IconListAdapter> task,
                                   IconListAdapter value) {
        try {
            this.grid.setAdapter(value);
        } catch (Exception e) {
            Toast.makeText(this.getActivity(), R.string.error_icons, Toast.LENGTH_SHORT).show();
        }
    }

}

File: ActivityLauncherApp/src/main/java/de/szalkowski/activitylauncher/IconPickerFragment.kt
package de.szalkowski.activitylauncher

import android.content.Context
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.AdapterView
import android.widget.AdapterView.OnItemClickListener
import android.widget.GridView
import android.widget.Toast
import androidx.fragment.app.Fragment

class IconPickerFragment : Fragment(), AsyncProvider.Listener<IconListAdapter> {
    private lateinit var grid: GridView

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        val view = inflater.inflate(R.layout.icon_picker, null)

        this.grid = view as GridView
        this.grid.onItemClickListener = OnItemClickListener { view, item, index, id -> Toast.makeText(activity, view.adapter.getItem(index).toString(), Toast.LENGTH_SHORT).show() }

        return view
    }

    override fun onAttach(activity: Context?) {
        super.onAttach(activity)

        val provider = IconListAsyncProvider(this.activity!!, this)
        provider.execute()
    }

    override fun onProviderFininshed(task: AsyncProvider<IconListAdapter>,
                                     value: IconListAdapter) {
        try {
            this.grid.adapter = value
        } catch (e: Exception) {
            Toast.makeText(this.activity, R.string.error_icons, Toast.LENGTH_SHORT).show()
        }

    }

}
